erts/emulator/beam/beam_emu.c

```diff
@@ -1499,7 +1501,11 @@ void process_main(Eterm * x_reg_array, FloatDef* f_reg_array)
      if (is_value(result)) {			\
 	 StoreBifResult(4, result);		\
      }						\
-     goto lb_Cl_error;				\
+     if (Arg(0) != 0) goto jump_f;		\
+     reg[0] = Op1;				\
```

Comment:
Not safe if a GC has been done in the arithmetic operation (if the error reason is `system_limit`, there could have been a GC).

Should be:

    reg[0] = reg[live]
    reg[1] = reg[live+1]